apps/client/lib/client/repeatable_lists/create_list_from_template.ex

```diff
@@ -22,56 +23,60 @@ defmodule Client.RepeatableLists.CreateListFromTemplate do
 
       {:ok, _result} =
         template
-        |> item_attrs(list)
-        |> Enum.reduce(Ecto.Multi.new(), fn item_cset, multi ->
-          Ecto.Multi.insert(multi, :rand.uniform(), item_cset)
+        |> section_attrs(list)
+        |> Enum.reduce(Ecto.Multi.new(), fn section_cset, multi ->
+          Ecto.Multi.insert(multi, :rand.uniform(), section_cset)
         end)
         |> Repo.transaction()
 
       {:ok, _result} =
         template
-        |> section_attrs(list)
-        |> Enum.reduce(Ecto.Multi.new(), fn section_cset, multi ->
-          Ecto.Multi.insert(multi, :rand.uniform(), section_cset)
+        |> item_attrs(list)
+        |> Enum.reduce(Ecto.Multi.new(), fn item_cset, multi ->
+          Ecto.Multi.insert(multi, :rand.uniform(), item_cset)
         end)
         |> Repo.transaction()
 
       list
     end)
   end
 
-  defp item_attrs(template, list) do
+  defp section_attrs(template, list) do
     from(
-      ti in TemplateItem,
-      where: ti.template_id == ^template.id,
+      ts in TemplateSection,
+      where: ts.template_id == ^template.id,
       select: %{
-        name: ti.name,
-        section_id: ti.section_id,
-        template_id: ti.template_id
+        name: ts.name,
+        template_id: ts.template_id,
+        template_section_id: ts.id
       }
     )
     |> Repo.all()
     |> Enum.map(fn attrs ->
       attrs
       |> Map.put(:list_id, list.id)
-      |> RepeatableLists.new_item_changeset()
+      |> RepeatableLists.new_section_changeset()
     end)
   end
 
-  defp section_attrs(template, list) do
+  defp item_attrs(template, list) do
     from(
-      ts in TemplateSection,
-      where: ts.template_id == ^template.id,
+      ti in TemplateItem,
+      where: ti.template_id == ^template.id,
+      left_join: ts in assoc(ti, :section),
+      left_join: s in Section,
+      on: s.template_section_id == ts.id,
```

Comment:
When creating these item attrs the only info we have is the template_id. To query for most of the attrs is easy because it's basically `template.items.map(&:attrs)`, but to get the section_id is a little more complex.

We know that the TemplateItem can belong to a TemplateSection. We also know that any Sections have a `template_section_id` field linking them to their parent TemplateSection. So what we do here is basically `template_item.template_section.section.id`